iceoryx_posh/CMakeLists.txt

```diff
@@ -245,9 +291,14 @@ target_include_directories(iceoryx_posh_roudi PRIVATE
     source/roudi
 )
 
+if(TOML_CONFIG)
+    set(POSH_CONFIG_LIB ${PROJECT_NAMESPACE}::iceoryx_posh_config)
```

Comment:
@elBoberido i did this here, because the linker in MacOS was complaining about missing references to iceoryx_posh_config.

Reply:
do we build the toml config on macOS? Nevertheless, `iceoryx_posh_roudi` should not depend on `iceoryx_posh_config` since the toml config might not be certified and then roudi needs to be build with a static config.
I think the dependency comes from the `CmdLineParser` maybe we need to move that somewhere else.

Reply:
it depends only on `iceoryx_posh_config` if TOML_CONFIG is set (Currently on per default). It also works when TOML Config is disabled, the variable is then just empty.  TOML Is also build in MacOS.

Reply:
this is only working because we don't build a RouDi with a static config. When `TOML_CONFIG` is `OFF` this is broken and won't work. If we want to keep that option, we maybe should create a test for it.

Reply:
I solved the dependency betweet `iceoryx_posh_roudi` and by moving some of the cmdline parser files to `posh_roudi` lib.
Then you have the choice to build either with or without TOML support. I added a RouDi with static config to iceperf example and let it build in Github CI.